pkg/agent/multicast/mcast_discovery.go

```diff
@@ -139,10 +139,14 @@ func (s *IGMPSnooper) processPacketIn(pktIn *ofctrl.PacketIn) error {
 		msg := igmp.(*protocol.IGMPv3MembershipReport)
 		for _, gr := range msg.GroupRecords {
 			mgroup := gr.MulticastAddress
-			klog.InfoS("Received IGMPv3 Report message", "group", mgroup.String(), "interface", iface.PodName)
+			klog.InfoS("Received IGMPv3 Report message", "group", mgroup.String(), "interface", iface.InterfaceName, "recordType", gr.Type, "sourceCount", gr.NumberOfSources)
+			evtType := groupJoin
+			if gr.Type == protocol.IGMPIsIn && gr.NumberOfSources == 0 {
```

Comment:
I think it should be `gr.Type == protocol.IGMPToIn`

Reply:
Do you mean we don't check gr.NumberOfSources? I think it is necessary, from rfc 3376 (IGMPv3),

Reply:
My bad. I forgot to add `gr.NumberOfSources == 0`. So the correct logic should be
```
if gr.Type == protocol.IGMPToIn && gr.NumberOfSources == 0 {
```

Reply:
I think your concern is correct, I would add support for CHANGE_TO_INCLUDE_MODE.